src/net_processing.cpp

```diff
@@ -2888,15 +2888,17 @@ bool SendMessages(CNode* pto, CConnman& connman, const std::atomic<bool>& interr
 
         // Address refresh broadcast
         int64_t nNow = GetTimeMicros();
-        if (!IsInitialBlockDownload() && pto->nNextLocalAddrSend < nNow) {
+        int64_t current_time = GetMockableTimeMicros(); // add a mockable time
+
+        if (!IsInitialBlockDownload() && pto->nNextLocalAddrSend < current_time) {
             AdvertiseLocal(pto);
-            pto->nNextLocalAddrSend = PoissonNextSend(nNow, AVG_LOCAL_ADDRESS_BROADCAST_INTERVAL);
+            pto->nNextLocalAddrSend = PoissonNextSend(current_time, AVG_LOCAL_ADDRESS_BROADCAST_INTERVAL);
         }
 
         //
         // Message: addr
         //
-        if (pto->nNextAddrSend < nNow) {
+        if (pto->nNextAddrSend < current_time) {
             pto->nNextAddrSend = PoissonNextSend(nNow, AVG_ADDRESS_BROADCAST_INTERVAL);
```

Comment:
I can't tell if this `nNow` should be replaced with `current_time` or not. If not, why not?

Reply:
You're right, screwed that up in a rebase! Thanks ❤️